gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/util/DefaultGraphManager.java

```diff
@@ -0,0 +1,201 @@
+     */
+    public Map<String, Graph> getGraphs() {
+        return graphs;
+    }
+
+    public Graph getGraph(String gName) {
+        return graphs.get(gName);
+    }
+
+    public void addGraph(String gName, Graph g) {
+        graphs.put(gName, g);
+    }
+
+    /**
+     * Get a list of the {@link TraversalSource} instances and their binding names as defined by Gremlin Server
+     * initialization scripts.
+     *
+     * @return a {@link Map} where the key is the name of the {@link TraversalSource} and the value is the
+     *         {@link TraversalSource} itself
+     */
+    public Map<String, TraversalSource> getTraversalSources() {
+        return traversalSources;
+    }
+
+    public TraversalSource getTraversalSource(String tsName) {
+        return traversalSources.get(tsName);
+    }
+
+    public void addTraversalSource(String tsName, TraversalSource ts) {
+        traversalSources.put(tsName, ts);
+    }
+
+    /**
+     * Get the {@link Graph} and {@link TraversalSource} list as a set of bindings.
+     */
+    public Bindings getAsBindings() {
+        final Bindings bindings = new SimpleBindings();
+        graphs.forEach(bindings::put);
+        traversalSources.forEach(bindings::put);
+        return bindings;
+    }
+
+    /**
+     * Rollback transactions across all {@link Graph} objects.
+     */
+    public void rollbackAll() {
+        graphs.entrySet().forEach(e -> {
+            final Graph graph = e.getValue();
+            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen())
+                graph.tx().rollback();
+        });
+    }
+
+    /**
+     * Selectively rollback transactions on the specified graphs or the graphs of traversal sources.
+     */
+    public void rollback(final Set<String> graphSourceNamesToCloseTxOn) {
+        closeTx(graphSourceNamesToCloseTxOn, Transaction.Status.ROLLBACK);
+    }
+
+    /**
+     * Commit transactions across all {@link Graph} objects.
+     */
+    public void commitAll() {
+        graphs.entrySet().forEach(e -> {
+            final Graph graph = e.getValue();
+            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen())
+                graph.tx().commit();
+        });
+    }
+
+    /**
+     * Selectively commit transactions on the specified graphs or the graphs of traversal sources.
+     */
+    public void commit(final Set<String> graphSourceNamesToCloseTxOn) {
+        closeTx(graphSourceNamesToCloseTxOn, Transaction.Status.COMMIT);
+    }
+
+    /**
+     * Basic graphManager has basic openGraph function.
+     */
+    public Graph openGraph(String graphName, Supplier<Graph> supplier) {
+        final Graph graph = graphs.get(graphName);
+        if (null != graph) {
+            return graph;
+        }
+        return supplier.get();
+    }
+
+    /**
+     *  Close graph
+     */
+    public void closeGraph(Graph graph) {
+        try {
+            graph.close();
```

Comment:
What does it mean to actually call `closeGraph()`? If "closed" does that mean it should still be in the available set of graphs? If so it would still be accessible to requests but it would be in a closed state. Different graphs will have different semantics with respect to what `close()` will do. Anyway, just posing the issue for thought at this point....not sure i have a strong opinion yet on what the behavior should be.